### ISSUE TITLE
Re: Wanted to show you project I made with your library

### ISSUE THREAD

**Issue by JohnBuhanan** [adrielcafe/voyager#46]
Hi @adrielcafe, I wanted to show you my sample app I made with Voyager: https://github.com/JohnBuhanan/MVI-Public

I studied a dozen github repos and felt like yours was closest to what I wanted.

Three things I thought you might find interesting:

1. I wrapped your Navigator so that I can do routing from ViewModels.
2. Each feature contains an `api` gradle module and an `impl` gradle module.  The `impl`s can be commented out from `settings.gradle` and `app.gradle` to unload them from Android Studio indexing and gradle builds.  This allows scaling for massively multi-module apps.
3. I used reflection to get access to the "internal factories" of ScreenRegistry.

Biggest thing I am thinking about now is best way to "navigate for result".

**Comment by ynsok** [comment]
Hi @JohnBuhanan  

Do you know why it is not working without reflection of ScreenRegistry?

**Comment by JohnBuhanan** [comment]
@ynsok 
I think there is type-erasure happening with the way I do Routing :(

Maybe there is a better way to do routing from ViewModel.

**Comment by ynsok** [comment]
Generally, I think it should be enough, and solve our problem. What do you think? 
@adrielcafe  @DevSrSouza  @JohnBuhanan 
```
     fun get(provider: ScreenProvider): Screen {
        val factory =
            factories[provider::class] ?: error("ScreenProvider not registered: ${provider::class.qualifiedName}")
        return factory(provider)
    }
```

**Comment by adrielcafe** [comment]
I've implemented the @ynsok suggestion in [1.0.0-beta16](https://github.com/adrielcafe/voyager/releases/tag/1.0.0-beta16), please reopen the issue if that was not enough.

About navigating from ViewModel/ScreenModel or any other class, I like the idea, some times that kind of logic is too complex and should not be in the UI layer. Feel free to open another issue about that, I'll start to think about an implementation.
